cmp/grid/GridModel.js

```diff
@@ -1031,6 +1032,93 @@ export class GridModel extends HoistModel {
         }
     }
 
+    /**
+     * Begin an inline editing session.
+     * @param {RecordOrId} [recOrId] - Record/ID to edit. If unspecified, the first selected Record
+     *      will be used, if any, or the first overall Record in the grid.
+     * @param {string} [colId] - ID of column on which to start editing. If unspecified, the first
+     *      editable column will be used.
+     * @return {Promise<void>}
+     */
+    async beginEditAsync({record, colId} = {}) {
+        const isReady = await this.whenReadyAsync();
```

Comment:
should we have a variant of this that throws if the grid cannot become ready?  

e.g. await this.ensureReadyAsync();

Reply:
As discussed, agree that would be a good addition - something we could look at along with #2551